API/main.py

```diff
@@ -0,0 +1,109 @@
+    filter=request.args.get("filter")
+    caffs = db.session.query(CAFF).all()
+    json_data = []
+    for caff in caffs:
+        if (not filter) or (filter in caff.name):
+            tmp = {"id" : caff.id,
+                    "name" : caff.name,
+                    "number_of_comments" : caff.comments}
+            json_data.append(tmp.copy())
+    return {"caffs" : json_data}
+
+@main.route('/caffs', methods=['POST'])
+@login_required
+def upload():
+    #prolly works
+    file= request.files['file']
+    caff = CAFF.query.filter_by(name=file.filename, uploader=current_user.id).first()
+    if caff or file.filename.split('.')[-1] != "caff":
+        return Response(response='File already uploaded', status=400)
+    new_caff = CAFF(url=upload_folder+"\\"+file.filename, name=file.filename, comments =0, uploader=current_user.id)
+    db.session.add(new_caff)
+    db.session.commit()
+    file.save(upload_folder+"\\"+file.filename)
+    return Response(response="Upload succesful", status=200)
+
+@main.route('/caffs/<caff_id>', methods=['GET'])
+@login_required
+def CAFFdetail(caff_id):
+    caff = db.session.query(CAFF).get(caff_id)
+    comments_list = []
+    comments = db.session.query(Comment).all()
+    for comment in comments:
+        if comment.caff_id == int(caff_id): #TODO if statment problem
+            tmp = {
+                "id" : comment.id,
+                "text" : comment.text,
+                "author" : db.session.query(User).get(comment.user_id).name
+            }
+            comments_list.append(tmp.copy())
+    uploader = db.session.query(User).get(caff_id) 
+    json_data = {
+        "id" : caff.id,
+        "preview_url" : caff.url, #wtf?
+        "comments" : comments_list,
+        "number_of_comments" : caff.comments,
+        "uploader" : {"id" : uploader.id ,"name" : uploader.name}
+    }
+    return json_data
+
+@main.route('/caffs/<caff_id>', methods=['DELETE'])
+@login_required
+def deleteCAFF(caff_id):
+    #TODO admin role
+    caff = db.session.query(CAFF).get(caff_id)
+    db.session.delete(caff)
+    comments = db.session.query(Comment).all()
+    for comment in comments:
+        if comment.caff_id == int(caff_id):
+            db.session.delete(comment)
+    db.session.commit()
+    return Response(status=200)
+
+@main.route('/caffs/<caff_id>/comments', methods=['POST'])
+@login_required
+def addComment(caff_id):
+    comment_text = request.json["text"]
+    comment = Comment(caff_id= caff_id, user_id=current_user.id, text=comment_text )
+    db.session.add(comment)
+    caff = db.session.query(CAFF).get(caff_id)
+    caff.comments += 1
+    db.session.commit()
+    return Response(status=200)
+
+@main.route('/caffs/<caff_id>/comments/<comment_id>', methods=['DELETE'])
+@login_required
+def deleteComment(caff_id, comment_id):
+    caff = db.session.query(CAFF).get(caff_id)
+    comment = db.session.query(Comment).get(comment_id)
+    db.session.delete(comment)
+    caff.comments -= 1
```

Comment:
isn't it `number_of_comments`? Or this is just DB name?

Reply:
just DB field name:: 
![image](https://user-images.githubusercontent.com/41287236/143836289-c2c55ddf-ac41-4756-b822-f46f88ddb40d.png)